### PR TITLE
feat: support services in Java SDK

### LINKED ISSUE #6449: Support services in Java SDK
We want to support users passing in `service_name` and `service_version` in the Java SDK

Example of Ruby SDK: https://github.com/highlight/highlight/pull/6229

Acceptance Criteria:
- can see service name in logs of Java app
- can see service version in logs of Java app
- backend errors are created with service name and version associated

### PR DESCRIPTION
**Pull request by neo773**
adds `service_name` to Java SDK

Screenshot 2023-09-22 at 2 23 00 AM

Screenshot 2023-09-22 at 2 23 13 AM

/closes #6449
/claim #6449

### CODE REVIEW

File: sdk/highlight-java/highlight-common/src/main/java/io/highlight/sdk/common/HighlightOptions.java [hunk 1]
```diff
@@ -9,7 +9,7 @@
  * Represents the options for highlighting in a project, including project ID,
  * backend URL, environment, version, and default attributes.
  */
-public record HighlightOptions(String projectId, String backendUrl, String enviroment, String version, boolean metric,
+public record HighlightOptions(String projectId, String backendUrl, String enviroment, String version, String serviceName, boolean metric,
```
Comment (Vadman97) [inline sdk/highlight-java/highlight-common/src/main/java/io/highlight/sdk/common/HighlightOptions.java:12]: could you add a `serviceVersion` as well per the [docs](https://www.highlight.io/docs/general/product-features/general-features/services)?

File: sdk/highlight-java/highlight-common/src/main/java/io/highlight/sdk/common/HighlightOptions.java
Comment (Vadman97) [inline sdk/highlight-java/highlight-common/src/main/java/io/highlight/sdk/common/HighlightOptions.java:1]: we'll need to bump the version of the SDK https://github.com/highlight/highlight/blob/36fa7cbcb5f0017c100ea2525575b36b44913a03/sdk/highlight-java/pom.xml#L40

could you make this `1.0.0-b1`?

File: sdk/highlight-java/highlight-common/src/main/java/io/highlight/sdk/common/HighlightOptions.java [hunk 2]
```diff
@@ -88,6 +90,19 @@ public Builder version(String version) {
...
+		 * If not set, the default value is <b>"unknown"</b>.
+		 *
+		 * @param serviceName the service name to set
+		 * @return this builder
+		 */
+		public Builder serviceName(String serviceName) {
+			this.serviceName = serviceName;
+			return this;
+		}
+
```
Comment (Vadman97) [inline sdk/highlight-java/highlight-common/src/main/java/io/highlight/sdk/common/HighlightOptions.java:105]: can we document the new behavior in https://github.com/highlight/highlight/blob/36fa7cbcb5f0017c100ea2525575b36b44913a03/docs-content/sdk/java.md#L4?

**Review by Vadman97 — APPROVED** [review]
thank you for your contribution!

**Comment by neo773** [comment]
@Vadman97 Pushed changes

**Pull request merged**